File: cpp/_34.cpp
// O(logN) / O(1)
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int l = bisect_left(nums, target);
        if (l == nums.size() || nums[l] != target)
            return {-1, -1};
        return {l, bisect_left(nums, target + 1) - 1};
    }
private:
    int bisect_left(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        while (l < r) {
            int m = l + (r - l) / 2;
            if (nums[m] >= target) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return l;
    }
};

File: python/_34.py
# O(logN) / O(1)
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l = self.bisect_left(nums, target)
        if l == len(nums) or nums[l] != target:
            return [-1, -1]
        return [l, self.bisect_left(nums, target + 1) - 1]
        
    def bisect_left(self, nums, target):
        l, r = 0, len(nums)
        while l < r:
            m = l + (r - l) // 2
            if nums[m] >= target:
                r = m
            else:
                l = m + 1
        return l